discord/embeds.py

```diff
@@ -152,15 +106,53 @@ class EmbedFooter(EmbedProxy):
     def __init__(
         self,
         text: str,
-        icon_url: MaybeEmpty[str] = EmptyEmbed,
-        proxy_icon_url: MaybeEmpty[str] = EmptyEmbed,
+        icon_url: str | None = None,
     ) -> None:
-        layer = {
-            k: v
-            for k, v in locals().items()
-            if k in {"text", "icon_url", "proxy_icon_url"} and v is not EmptyEmbed
+        self.text: str = text
+        self.icon_url: str | None = icon_url
+        self.proxy_icon_url: str | None = None
+
+    @classmethod
+    def from_dict(cls, data: dict[str, str | None]) -> EmbedFooter:
+        self = cls.__new__(cls)
+        self.text = data.get("text")
+        self.icon_url = data.get("icon_url")
+        self.proxy_icon_url = data.get("proxy_icon_url")
+        return self
+
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            "text": str(self.text),
+            "icon_url": str(self.icon_url),
```

Comment:
Same as comment for `EmbedAuthor.to_dict`
```suggestion
            "text": self.text,
            "icon_url": self.icon_url,
```

Reply:
see https://github.com/Pycord-Development/pycord/pull/2063#discussion_r1190304216